controller/src/main/java/org/jboss/as/controller/parsing/DeferredExtensionContext.java

```
@@ -126,7 +126,7 @@ private XMLStreamException loadModule(final String moduleName, final XMLMapper x
             }
             return null;
         } catch (final ModuleLoadException e) {
-            throw ControllerLogger.ROOT_LOGGER.failedToLoadModule(e);
+            return ControllerLogger.ROOT_LOGGER.failedToLoadModule(e);
```

Could you also explain the motivation for this change?

The original code intention was to throw the exception instead of returning it as the invocation result, so, I don't get the reason for changing it now. In addition, is a change unrelated to the original issue intention, I don't see it connected.

---

@yersan It's related. The caller checks for the existence of the exception and throws it:
* [https://github.com/wildfly/wildfly-core/blob/62da6381bbbd5c5305e36bc25a03834f265d8b4e/controller/src/main/java/org/jboss/as/controller/parsing/DeferredExtensionContext.java#L82](https://github.com/wildfly/wildfly-core/blob/62da6381bbbd5c5305e36bc25a03834f265d8b4e/controller/src/main/java/org/jboss/as/controller/parsing/DeferredExtensionContext.java#L82)
* [https://github.com/wildfly/wildfly-core/blob/62da6381bbbd5c5305e36bc25a03834f265d8b4e/controller/src/main/java/org/jboss/as/controller/parsing/DeferredExtensionContext.java#L93](https://github.com/wildfly/wildfly-core/blob/62da6381bbbd5c5305e36bc25a03834f265d8b4e/controller/src/main/java/org/jboss/as/controller/parsing/DeferredExtensionContext.java#L93)

In the current implementation it is always null and therefor the if evaluates always to false.

---

I see thanks, so it's a matter of removing the `if` and kept the exception be thrown or return the exception and keeping the `if`

The former looks cleaner to me; less code to read :)

In any case, since it is a private method both options would work fine. If finally, you choose for the second one, returning the exception, you have also to remove the throws clause at the method level.

---

@yersan I have switched to the variant removing the `if`, please have a look. Tests passed except win full.